Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -22,9 +22,8 @@ namespace Microsoft.MixedReality.Toolkit.Services.InputSystem
     public class FocusProvider : BaseService, IMixedRealityFocusProvider
     {
         private readonly HashSet<PointerData> pointers = new HashSet<PointerData>();
-        private readonly HashSet<GameObject> pendingOverallFocusEnterSet = new HashSet<GameObject>();
-        private readonly HashSet<GameObject> pendingOverallFocusExitSet = new HashSet<GameObject>();
-        private readonly List<PointerData> pendingPointerSpecificFocusChange = new List<PointerData>();
+        private readonly Dictionary<GameObject, HashSet<IMixedRealityPointer>> overallFocusSet = new Dictionary<GameObject, HashSet<IMixedRealityPointer>>();
```

Comment:
Could we just keep the count of pointers over each object instead?

Reply:
The idea of an AR Indiana Jones reaching out for the golden Trophy just came to be. I think you want to be able to make sure that both hands and not just a hand and the gaze are reaching out to it before you throw in the glow shader :)